astropy/table/tests/test_table.py

```diff
@@ -2693,3 +2694,14 @@ def test_items():
 
     for i in list(t.items()):
         assert isinstance(i, tuple)
+
+
+def test_overwrite_serialized_meta():
+    # This used to cause an error because the meta data table
+    # was not removed from the existing file.
+
+    with h5py.File('test_data.h5', 'w', driver='core', backing_store=False) as out:
+        t = table.Table([[1, 2, 3]])
+
+        t.write(out, path='data', serialize_meta=True)
+        t.write(out, path='data', serialize_meta=True, append=True, overwrite=True)
```

Comment:
Ideally, make this a test with two tables and ones that have meta data, and then check that the table and metadata are in fact replaced. in `test_hdf5`, `test_read_write_existing_overwrite` and `test_preserve_serialized` might be good examples to use for making a new test.